webapp-django/crashstats/crashstats/views.py

```diff
@@ -1354,9 +1357,156 @@ def exploitable_crashes(
     ))
     context['version'] = versions
     context['report'] = 'exploitable'
+    context['new_link_url'] = reverse('crashstats:exploitability_report')
+    query_string = {'product': product}
+    if versions:
+        query_string['version'] = versions
+    context['new_link_url'] += '?' + urllib.urlencode(query_string, True)
     return render(request, 'crashstats/exploitability.html', context)
 
 
+@utils.json_view
+@pass_default_context
+@permission_required('crashstats.view_exploitability')
+def exploitability_report(request, default_context=None):
+    context = default_context or {}
+
+    if not request.GET.get('product'):
+        url = reverse('crashstats:exploitability_report')
+        url += '?' + urllib.urlencode({
+            'product': settings.DEFAULT_PRODUCT
+        })
+        return redirect(url)
+
+    available_products = {}
+    for product, versions in context['releases'].items():
+        if product not in available_products:
+            available_products[product] = []
+        for version_info in versions:
+            available_products[product].append(version_info['version'])
+
+    form = forms.ExploitabilityReportForm(
+        request.GET,
+        available_products=available_products,
+    )
+    if not form.is_valid():
+        return http.HttpResponseBadRequest(str(form.errors))
+
+    if request.GET.get('format') == 'json':
+        product = form.cleaned_data['product']
+        version = form.cleaned_data['version']
+
+        api = SuperSearchUnredacted()
+        # XXX ADRIAN, how do I use settings.EXPLOITABILITY_BATCH_SIZE
+        # to get a limited set?
+        params = {
+            'product': product,
+            'version': version,
+            '_results_number': 0,
+            '_aggs.signature': 'exploitability',
```

Comment:
So what's going to happen here is that the signatures will be sorted by their number of occurrences in crashes. It's basically going to be topcrashers, but you'll have the exploitability of each signature with it. I don't know if that's what we want, or if we want to see the most "exploitable" crashes first, whatever that is.